packages/lib-classifier/src/plugins/tasks/text/components/TextTask/TextTask.spec.js

```diff
@@ -1,247 +1,299 @@
-        expect(annotation.value).to.equal(expectedText)
-      })
+    it('should have a labelled text input', function () {
+      expect(textInput).to.exist()
     })
-    
-    describe('text tagging (special case for literal insertions)', function () {
-      beforeEach(function () {
-        annotation.update('Dungeons and Dragons')
-        wrapper = mount(
-          <TextTask
-            annotation={annotation}
-            task={task}
-          />,
-          {
-            wrappingComponent: Grommet,
-            wrappingComponentProps: { theme: zooTheme }
-          }
-        )
-        const ampersandButton = wrapper.find('button').find({ value: '&' })
-        const fakeEvent = {
-          currentTarget: {
-            value: '&'
-          }
+
+    it('should save typed text', function () {
+      expect(annotation.value).to.equal('Hello world!')
+    })
+
+    it('should display the current annotation', function () {
+      expect(textInput.value).to.equal(annotation.value)
+    })
+
+    it('should have text modifier buttons', function () {
+      expect(modifiers).to.exist()
+    })
+
+    it('should have a modifier button for each text tag', function () {
+      expect(buttons).to.have.lengthOf(task.text_tags.length)
+    })
+  })
+
+  describe('tagging text', function () {
+    let buttons, modifiers, textInput
+
+    const task = Task.TaskModel.create({
+      instruction: 'Type something here',
+      taskKey: 'T0',
+      text_tags: ['insertion', 'deletion', '&'],
+      type: 'text'
+    })
+    const annotation = task.defaultAnnotation()
+
+    before(async function () {
+      const user = userEvent.setup()
+      render(
+        <TextTask
+          annotation={annotation}
+          task={task}
+        />,
+        {
+          wrapper: withGrommet
         }
-        const textArea = wrapper.find(TextArea).getDOMNode()
-        textArea.selectionStart = 9
-        textArea.selectionEnd = 12
-        ampersandButton.simulate('click', fakeEvent)
+      )
+      modifiers = screen.getByRole('group', {
+        name: 'TextTask.TextTagButtons.modifiers'
       })
-
-      it('should insert literal tag text', function () {
-        const textArea = wrapper.find(TextArea).getDOMNode()
-        const expectedText = 'Dungeons & Dragons'
-        const updatedText = textArea.value
-        expect(updatedText).to.equal(expectedText)
+      const insertion = within(modifiers).getByRole('button', {
+        name: 'TextTask.TextTagButtons.modifiers insertion'
       })
-
-      it('should save the modified text', function () {
-        const expectedText = 'Dungeons & Dragons'
-        expect(annotation.value).to.equal(expectedText)
+      textInput = screen.getByRole('textbox', {
+        name: task.instruction
       })
+      await user.type(textInput, 'Hello, this is some test text.')
+      await user.pointer([
+        {
+          target: textInput,
+          offset: 7,
+          keys: '[MouseLeft>]'
```

Comment:
Oh wow, I did not realise [this](https://testing-library.com/docs/user-event/pointer#pressing-a-button-or-touching-the-screen) is how you action to define "mouse down (and then drag)"

Minor question: would you also need to add the "mouse up" action (`[/MouseLeft]`) after you move to offset 11?

Reply:
Maybe. To be honest, I can’t remember if you need to explicitly add that here. It’s a good question.